postgres/tests/test_relationsmanager.py

```diff
@@ -10,6 +10,48 @@
 pytestmark = pytest.mark.unit
 
 
+@pytest.mark.parametrize(
+    'relations_config,expected_filter',
+    [
+        (
+            [
+                {'relation_regex': 'ix.*', 'schemas': ['public', 's1', 's2']},
+                {'relation_regex': 'ibx.*', 'schemas': ['public']},
+                {'relation_regex': 'icx.*', 'schemas': ['public']},
+            ],
+            "( relname ~ 'ix.*' AND schemaname = ANY(array['public','s1','s2']::text[]) ) "
+            "OR ( relname ~ 'ibx.*' AND schemaname = ANY(array['public']::text[]) ) "
+            "OR ( relname ~ 'icx.*' AND schemaname = ANY(array['public']::text[]) )",
+        ),
+        (
+            [
+                {'relation_regex': '.+_archive'},
+            ],
+            "( relname ~ '.+_archive' )",
+        ),
+        (
+            [
+                {'relation_name': 'my_table', 'schemas': ['public', 'app'], 'relkind': ['r']},  # relkind ignored
```

Comment:
A bit confused about the tests. Is the expected behavior of this fix supposed to be that config parsing takes in all values in relations list (i.e. both `relation_name` and `relation_regex`)? In that case why is it expected that `relkind` is ignored? Sorry not too familiar with relations config 😅

Reply:
The purpose of these tests is to start adding some more real-world cases and boundary cases that customers might actually set up.

`relkind` is ignored in this test due to some hacky behavior that checks the contents of the template query: https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/relationsmanager.py#L273. I could add `FROM pg_locks` to the query and it will be used, but that also changes the test.

I'd like to make that more explicitly set, but I'm also trying to keep changes small to address the bug in question without changing APIs. So my intention of including `relkind` in the configs here is to start to add coverage for these more complex configs that will fail when those parts are touched.

Reply:
Ah gotcha. Could you leave a note for the `relkind` options about the hacky behavior in case that's changed in the future?